src/Symfony/Component/Notifier/Bridge/Lox24/Lox24Transport.php

```diff
@@ -0,0 +1,201 @@
+                    $from
+                )
+            );
+        }
+
+
+        $body = [
+            'sender_id' => $from,
+            'phone' => $message->getPhone(),
+            'text' => $message->getSubject(),
+        ];
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $body = $this->setIsTextDeleted($body, $options);
+        $body = $this->setCallbackData($body, $options);
+        $body = $this->setDeliveryAt($body, $options);
+        $body = $this->setServiceCode($body, $options);
+        $body = $this->setVoiceLang($body, $options);
+
+        $response = $this->client->request('POST', sprintf("https://%s/sms", $this->getEndpoint()), [
+            'headers' => [
+                'X-LOX24-AUTH-TOKEN' => $this->auth,
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+                'User-Agent' => 'LOX24 Symfony Notifier',
+            ],
+            'body' => $body,
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote LOX24 server.', $response, 0, $e);
+        }
+
+        if (201 !== $statusCode) {
+            $error = $response->toArray(false);
+
+            throw new TransportException(
+                sprintf("Unable to send the SMS: %s", $error['detail']),
+                $response
+            );
+        }
+
+        $success = $response->toArray(false);
+
+        $sentMessage = new SentMessage($message, (string)$this);
+        $sentMessage->setMessageId($success['uuid']);
+
+        return $sentMessage;
+    }
+
+    private function isFromValid(string $from): bool
+    {
+        return preg_match('/^[.\-a-zA-Z0-9_ ]{2,11}$/', $from) || preg_match('/^\+[1-9]\d{1,14}$/', $from);
+    }
+
+    private function setIsTextDeleted(array $body, array $options): array
+    {
+        $body['is_text_deleted'] = (bool)($options['is_text_delete'] ?? false);
+
+        return $body;
+    }
+
+    private function setCallbackData(array $body, array $options): array
+    {
+        if (!empty($options['callback_data'])) {
+            $body['callback_data'] = $options['callback_data'];
+        }
+
+        return $body;
+    }
+
+    private function setDeliveryAt(array $body, array $options): array
+    {
+        $body['delivery_at'] = max((int)($options['delivery_at'] ?? 0), 0);
+
+        return $body;
+    }
+
+    private function setServiceCode(array $body, array $options): array
+    {
+        $code = $options['type'] ?? Type::Sms->value;
+        $type = Type::tryFrom((string)$code);
+
+        if (!$type) {
+            throw new InvalidArgumentException(sprintf("Invalid type: %s", $code));
+        }
+
+        $body['service_code'] = $this->getServiceCode($type)->value;
```

Comment:
instead of a separate ServiceCode enum that is never used for any type safety thing (defeating the purpose of using an enum), I would suggest implementing a `getServiceCode` method in the `Type` enum looking like that:

```php
public function getServiceCode(): string
{
    return match ($this) {
        case self::Sms => 'direct',
        case self::Voice => 'text2speech',
    };
}
```

Reply:
done

Reply:
I am curious @stof, I've never seen the case before the self:: in the match()

Is this a real thing or just a typo in your dummy code?

Reply:
That's a typo in my dummy code (mixing the syntax of `switch` and `match`)